scripts/quests/ahtUrhgan/Divine_Interference.lua

```diff
@@ -0,0 +1,235 @@
+        rewardMask = utils.mask.setBit(rewardMask, 4, true)
+    end
+
+    return rewardMask
+end
+
+local function giveQuestReward(player, eventOption)
+    local wasRewarded = true
+
+    if eventOption <= 3 then
+        wasRewarded = npcUtil.giveItem(player, rewardItems[eventOption - 1])
+    elseif eventOption == 4 then
+        npcUtil.giveCurrency(player, 'gil', 10000)
+    elseif eventOption == 5 then
+        player:addSpell(xi.magic.spell.ALEXANDER)
+        player:messageSpecial(ahtUrhganID.text.ALEXANDER_UNLOCKED, 0, 0, 1)
+    end
+
+    return wasRewarded
+end
+
+quest.sections =
+{
+    -- Section: Quest available
+    {
+        check = function(player, status, vars)
+            return status == xi.questStatus.QUEST_AVAILABLE and
+                player:hasCompletedQuest(xi.questLog.AHT_URHGAN, xi.quest.id.ahtUrhgan.WAKING_THE_COLOSSUS) and
+                player:getCurrentMission(xi.mission.log_id.TOAU) == xi.mission.id.toau.ETERNAL_MERCENARY
+        end,
+
+        [xi.zone.AHT_URHGAN_WHITEGATE] =
+        {
+            ['Imperial_Whitegate'] =
+            {
+                onTrigger = function(player, npc)
+                    if quest:getVar(player, 'Prog') == 1 then
+                        return quest:progressEvent(931, 1, 0, 0, 2952354, 1, 2186, 4, 0, 0)
+                    end
+                end,
+            },
+
+            onTriggerAreaEnter =
+            {
+                [10] = function(player, triggerArea)
+                    if quest:getVar(player, 'Prog') == 0 then
+                        return quest:progressEvent(930, 1, 1, 255, 0, 67108863, 165010213, 0, 0, 0)
+                    end
+                end,
+            },
+
+            onEventFinish =
+            {
+                [930] = function(player, csid, option, npc)
+                    quest:setVar(player, 'Prog', 1)
+                end,
+
+                [931] = function(player, csid, option, npc)
+                    if option == 1 then
+                        quest:begin(player)
+                    end
+                end,
+            },
+        },
+    },
+
+    -- Section: Quest accepted
+    {
+        check = function(player, status, vars)
+            return status == xi.questStatus.QUEST_ACCEPTED
+        end,
+
+        [xi.zone.ALZADAAL_UNDERSEA_RUINS] =
+        {
+            ['blank_lamp'] =
+            {
+                onTrigger = function(player, npc)
+                    if
+                        quest:getVar(player, 'Prog') == 2 and
+                        player:hasKeyItem(xi.ki.LIGHTNING_CELL)
+                    then
+                        return quest:progressEvent(308)
+                    end
+                end,
+            },
+
+            onZoneIn =
+            {
+                function(player, prevZone)
+                    if quest:getVar(player, 'Prog') == 4 then
+                        return 309 -- correct cs from capture but doesn't play for some reason?
```

Comment:
Can't figure out why this cutscene doesn't work, tried passing it more params but doesn't seem to make a difference. Tried !cs and !exec player:startEvent. Event can be seen in a capture here: https://www.youtube.com/watch?v=fkIgQxGs9qE&t=2498s

Reply:
This also seems to affect the same CS for Waking the Colossus which is event 307.

Reply:
Was thinking about this.  Are there packet caps from the fight? I wonder if there's a text table change or event flag that needs to be set to support displaying the event.

Reply:
I stayed up late with him one night trying to get this right. We emulated every packet exactly as the capture and verified with Captain that they were the same. 

My best guess is some npc doesn't have the right properties and is causing the cutscenes to not proceed, but we tweaked every one in the capture to have exactly the same values and still would black screen. 

Also I highly suspect amp moved on, I haven't heard from him in months

Reply:
I looked at this the other day coincidentally and I'm more or less thinking the same thing, I'm leaning towards needing the 0x00E entity update packet code fixed before looking more into this